controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # Velocity controllers for each joint
    front_left_steering_controller:
      type: velocity_controllers/JointGroupVelocityController

    front_right_steering_controller:
      type: velocity_controllers/JointGroupVelocityController

    rear_left_steering_controller:
      type: velocity_controllers/JointGroupVelocityController

    rear_right_steering_controller:
      type: velocity_controllers/JointGroupVelocityController

    front_left_wheel_controller:
      type: velocity_controllers/JointGroupVelocityController

    front_right_wheel_controller:
      type: velocity_controllers/JointGroupVelocityController

    rear_left_wheel_controller:
      type: velocity_controllers/JointGroupVelocityController

    rear_right_wheel_controller:
      type: velocity_controllers/JointGroupVelocityController


# Controller Configurations
front_left_steering_controller:
  ros__parameters:
    joints:
      - front_left_steering_joint

front_right_steering_controller:
  ros__parameters:
    joints:
      - front_right_steering_joint

rear_left_steering_controller:
  ros__parameters:
    joints:
      - rear_left_steering_joint

rear_right_steering_controller:
  ros__parameters:
    joints:
      - rear_right_steering_joint

front_left_wheel_controller:
  ros__parameters:
    joints:
      - front_left_wheel_joint

front_right_wheel_controller:
  ros__parameters:
    joints:
      - front_right_wheel_joint

rear_left_wheel_controller:
  ros__parameters:
    joints:
      - rear_left_wheel_joint

rear_right_wheel_controller:
  ros__parameters:
    joints:
      - rear_right_wheel_joint
